src/cromshell/status/command.py

```diff
@@ -114,34 +104,48 @@ def main(config, workflow_id):
-        for instanceDic in workflow_metadata["calls"][call]:
-            execution_statuses.append(instanceDic["executionStatus"])
-
-        # Create a key and value pair for the execution status and number of times
-        # it is seen
-        status_count = Counter(execution_statuses)
-        # For each status for the call add the call name and status name and count
-        # to list
-        for status in set(execution_statuses):
-            status_element = f'{{"{status}": {status_count[status]}}}'
-            tmp_execution_status.append(f"{call_element}{status_element}")
-
-    tmp_execution_status_json = ", ".join(tmp_execution_status)
-    tmp_execution_status_json = "{" + tmp_execution_status_json + "}"
-    return [json.loads(tmp_execution_status_json)]
+def check_for_failure(metadata: dict):
+    """Checks a workflow metadata dictionary for failing statuses
+    Returns True to indicate workflow or some task(s) has failed"""
+    workflow_failed = False
+    # If the given dictionary contains a 'status' key and has
+    # value of "Failed" then exit the function returning
+    # True to indicate workflow has failed
+    if "status" in metadata:
+        if metadata["status"] == "Failed":
+            workflow_failed = True
+            return workflow_failed
+    # If the dictionary does not contain a failed value for
+    # its status key or if status key does does not exist
+    # then we'll iterate through this loop in search for
+    # task status or subworkflows.
+    for key, value in metadata.items():
+        # If a dictionary value is encountered then recursion is used iterate through
+        # the dictionary to find task statuses. The reason for this is due to the tree
+        # structure of the metadata dictionary which holds the calls, subworkflow calls,
+        # and their status within layers of the dictionary. Depth First Search recursion
+        # is used to traverse the dictionary by iterating through any dictionary
+        # this function comes across first and reporting whether it found a failure
+        if isinstance(value, dict):
+            workflow_failed = check_for_failure(value)
+
+        # If a list value is encountered then the dictionary being traversed through is
+        # a key and value pair where the key is the task name and the value is the
+        # number of times the task was executed (shard), which is normally 1 unless the
+        # task was scattered. Hence the value for the key is a list to account
+        # for all the shards for a task. Each item(shard) within this list is a
+        # dictionary holding status for the shard.
+        # We'll want to check each shard to determine whether it's status has
+        # Failed
+        if isinstance(value, list):
+            for i, shard in enumerate(value):
+                if "subWorkflowMetadata" in shard.keys():
+                    workflow_failed = check_for_failure(shard["subWorkflowMetadata"])
```

Comment:
Does this work?  It seems like a failing workflow followed by a successful one in the enumeration will overwrite the return value.  Am I crazy?

Reply:
You're right, i removed the `workflow_failed` variable and instead the function will immediately return `True` if it encounters a failed workflow or task.